### PR TITLE
Add option to prefer CloudFront compression supported MIME types

### LINKED ISSUE #53: Incorrect MIME type for JS assets (CloudFront)
The default MIME type for .js files is resolved as `application/ecmascript` but CloudFront requires `application/javascript` to be able to automatically compress the responses.

### PR DESCRIPTION
**Pull request by exoszajzbuk**
Fixes #53

### CODE REVIEW

File: lib/capistrano/s3/publisher.rb [hunk 1]
```diff
@@ -189,17 +192,26 @@ def self.has_gzipped_version?(file)
...
         end
 
-        def self.mime_type_for_file(file)
+        def self.mime_type_for_file(file, prefer_cf_mime_types)
           type = MIME::Types.type_for(file)
+
+          if prefer_cf_mime_types
+            intersection = type & Capistrano::S3::MIMETypes::CF_MIME_TYPES
+
+            if !intersection.empty?
```
Comment (j15e) [inline lib/capistrano/s3/publisher.rb]: I would avoir negated conditions, maybe 
```suggestion
            unless intersection.empty?
```

File: lib/capistrano/s3/publisher.rb [hunk 2]
```diff
@@ -189,17 +192,26 @@ def self.has_gzipped_version?(file)
...
+              type = intersection
```
Comment (j15e) [inline lib/capistrano/s3/publisher.rb]: I would assign first type here 
```suggestion
              type = intersection.first
```

File: lib/capistrano/s3/publisher.rb [hunk 3]
```diff
@@ -189,17 +192,26 @@ def self.has_gzipped_version?(file)
...
+            end
+          end
+
           (type && !type.empty?) ? type[0] : nil
```
Comment (j15e) [inline lib/capistrano/s3/publisher.rb]: could be just `type` with .first up there

```suggestion
          type
```

**Comment by exoszajzbuk** [comment]
@j15e can you please have a look at, and if all fine, release a new gem version?

**Comment by j15e** [comment]
Done! https://rubygems.org/gems/capistrano-s3/versions/2.4.0

**Pull request merged**